lib/Target/Lattigo/LattigoEmitter.cpp

```diff
@@ -619,8 +691,28 @@ LogicalResult LattigoEmitter::printOperation(CKKSEncodeOp op) {
   os << maxSlotsName << ")\n";
   os << "for i := range " << packedName << " {\n";
   os.indent();
-  os << packedName << "[i] = float64(" << getName(op.getValue()) << "[i \% len("
-     << getName(op.getValue()) << ")])\n";
+  if (getElementTypeOrSelf(op.getValue().getType()).getIntOrFloatBitWidth() ==
+      1) {
+    // if value[i] {
```

Comment:
I wonder if you could make this a single const string in an anon namespace and use a format var for the name? you could share the encode strings, and maybe also the negate could use a const string

Reply:
Made the boolToI64/F64 template string and also the negate template string.